src/store/api/index.ts

```diff
@@ -22,7 +23,21 @@ export const apiSlice = createApi({
               method: 'POST',
               body: credentials,
             }),
+
+            
           }),
+          getAllRideRequests: builder.query<RideRequest[], void>({
+            query: () => '/ride-requests',
+          }), 
+      
+        searchRequestsByName: builder.query<RideRequest[], string>({
```

Comment:
searchRequestsByName and filterByStatus are provided as a single endpoint..you should combine the endpoints into one